### ISSUE TITLE
Bug: Wrong cooked value of `TemplateElement` in `TaggedTemplateExpression `

### ISSUE THREAD

**Issue by fisker** [typescript-eslint/typescript-eslint#11353]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&showAST=es&fileType=.tsx&code=MoFwTglgdg5gdGAhgdwAYB0CuANX3VA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
String.raw`\uXXXX`
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

`TemplateElement.value.cooked` should be `null`.

### Actual Result

`TemplateElement.value.cooked` is "\uXXXX".

### Versions

| `@typescript-eslint/typescript-estree` | `8.35.0` |

**Comment by bradzacher** [comment]
Next time it would be good to link the justification as to why this should be null rather than just stating that it should be null.

https://github.com/estree/estree/blob/2bc9ea235184b6164e31b1088575447657e686c6/es2018.md?plain=1#L34

**Comment by fisker** [comment]
I did search `TaggedTemplateExpression` in ESTree, but didn't see this one, only found [the one in ES2015](https://github.com/estree/estree/blob/2bc9ea235184b6164e31b1088575447657e686c6/es2015.md#templateelement)...
